modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts

```diff
@@ -230,4 +234,73 @@ export class EcdsaUtils extends baseTSSUtils<KeyShare> {
       recipientIndex === 1 ? await keychains.add(userKeychainParams) : await keychains.createBackup(userKeychainParams);
     return result;
   }
+
+  /**
+   * Signs the transaction associated to the transaction request.
+   * @param {string | TxRequest} params.txRequest - transaction request object or id
+   * @param {string} params.prv - decrypted private key
+   * @param { string} params.reqId - request id
+   * @returns {Promise<TxRequest>} fully signed TxRequest object
+   */
+  async signTxRequest(params: {
+    txRequest: string | TxRequest;
+    prv: string;
+    reqId: IRequestTracer;
+  }): Promise<TxRequest> {
+    let txRequestResolved: TxRequest;
+    let txRequestId: string;
+
+    const { txRequest, prv } = params;
+
+    if (typeof txRequest === 'string') {
+      txRequestResolved = await getTxRequest(this.bitgo, this.wallet.id(), txRequest);
+      txRequestId = txRequestResolved.txRequestId;
+    } else {
+      txRequestResolved = txRequest;
+      txRequestId = txRequest.txRequestId;
+    }
+
+    const userSigningMaterial: ECDSAMethodTypes.SigningMaterial = JSON.parse(prv);
+    if (!userSigningMaterial.backupNShare) {
+      throw new Error('Invalid user key - missing backupNShare');
+    }
+    const MPC = new Ecdsa();
+    const signingKey = MPC.keyCombine(userSigningMaterial.pShare, [
+      userSigningMaterial.bitgoNShare,
+      userSigningMaterial.backupNShare,
+    ]);
+
+    const userSignShare = await ECDSAMethods.createUserSignShare(signingKey.xShare, signingKey.yShares[3]);
+
+    const bitgoToUserAShare = (await ECDSAMethods.sendShareToBitgo(
+      this.bitgo,
+      this.wallet.id(),
+      txRequestId,
+      SendShareType.KShare,
+      userSignShare.kShare
+    )) as AShare;
+
+    const userToBitGoGShare = await ECDSAMethods.createUserGammaShare(userSignShare.wShare, bitgoToUserAShare);
+
+    const bitgoToUserDShare = (await ECDSAMethods.sendShareToBitgo(
+      this.bitgo,
+      this.wallet.id(),
+      txRequestId,
+      SendShareType.MUShare,
+      userToBitGoGShare.muShare as ECDSA.MUShare
+    )) as DShare;
+
+    const userOShare = await ECDSAMethods.createUserOmicronShare(userToBitGoGShare.gShare as ECDSA.GShare);
```

Comment:
Here too, we are returning the user's omicron share as well as their delta share, can we rename the variable/function name to reflect this?